src/modules/file/dto/update-file-meta.dto.ts

```diff
@@ -1,11 +1,20 @@
 import { ApiProperty } from '@nestjs/swagger';
-import { IsString } from 'class-validator';
+import { IsOptional, IsString } from 'class-validator';
 
 export class UpdateFileMetaDto {
   @IsString()
   @ApiProperty({
     example: 'New name',
     description: 'The name the file is going to be updated to',
   })
-  plainName: string;
+  @IsOptional()
+  plainName?: string;
+
+  @IsString()
+  @ApiProperty({
+    example: 'New type',
+    description: 'The new type that the file is going to have',
+  })
+  @IsOptional()
+  type?: string;
 }
```

Comment:
since you are using .trim() in the service, you could move that to this DTO.

```
import { ApiProperty } from '@nestjs/swagger';
import { IsOptional, IsString, IsNotEmpty } from 'class-validator';
import { Transform } from 'class-transformer';

export class UpdateFileMetaDto {
  @IsOptional()
  @IsString()
  @IsNotEmpty({ message: 'Filename cannot be empty' })
  @Transform(({ value }) => value?.trim())
  @ApiProperty({
    example: 'New name',
    description: 'The name the file is going to be updated to',
  })
  plainName?: string;

  @IsOptional()
  @IsString()
  @IsNotEmpty({ message: 'File type cannot be empty' })
  @Transform(({ value }) => value?.trim())
  @ApiProperty({
    example: 'New type',
    description: 'The new type that the file is going to have',
  })
  type?: string;
}
```

Reply:
mmm not really as the trim() is done just to ckeck if its empty, not for the update property itself